sklearn/manifold/t_sne.py

```diff
@@ -336,6 +348,11 @@ def _gradient_descent(objective, p0, it, n_iter,
 
     tic = time()
     for i in range(it, n_iter):
+        check_convergence = (i + 1) % n_iter_check == 0
+        # only compute the error when needed
+        if 'compute_error' in kwargs:
+            kwargs['compute_error'] = check_convergence
```

Comment:
I think you need `or i == n_iter - 1` here so that the verbose-printed KL-divergence and that set in `kl_divergence_` is accurate.